docs/table_design/Temporary_partition.md

```diff
@@ -0,0 +1,311 @@
+
+```SQL
+ALTER TABLE site_access
+ADD TEMPORARY PARTITION tp3 VALUES LESS THAN ("2020-04-01")
+ ("replication_num" = "1")
+DISTRIBUTED BY HASH (site_id) BUCKETS 5;
+```
+
+Create multiple partitions at a time in the table `site_access` by using the `START (...) END (...) EVERY (...)` syntax, and specify the range of these partitions as `[2020-04-01, 2021-01-01)` with a monthly partition granularity.
+
+```SQL
+ALTER TABLE site_access 
+ADD TEMPORARY PARTITIONS START ("2020-04-01") END ("2021-01-01") EVERY (INTERVAL 1 MONTH);
+```
+
+### Usage notes
+
+- The partition column for a temporary partition must be the same as the partition column for the original formal partition based on which you create the temporary partition and cannot be changed.
+- The name of a temporary partition cannot be the same as the name of any formal partition or other temporary partition.
+- The ranges of all temporary partitions in a table cannot overlap, but the range of a temporary partition and a formal partition can overlap.
+
+## Show temporary partitions
+
+You can view the temporary partitions by using the [SHOW TEMPORARY PARTITIONS](../sql-reference/sql-statements/data-manipulation/SHOW%20PARTITIONS) command.
+
+```SQL
+SHOW TEMPORARY PARTITIONS FROM [db_name.]table_name [WHERE] [ORDER BY] [LIMIT]
+```
+
+## Load data into temporary partitions
+
+You can load data into one or more temporary partitions by using the `INSERT INTO` command, STREAM LOAD, or BROKER LOAD.
+
+### Load data by using `INSERT INTO` command
+
+Example:
+
+```SQL
+INSERT INTO site_access TEMPORARY PARTITION (tp1) VALUES ("2020-01-01",1,"ca","lily",4);
+INSERT INTO site_access TEMPORARY PARTITION (tp2) SELECT * FROM site_access_copy PARTITION p2;
+INSERT INTO site_access TEMPORARY PARTITION (tp3, tp4,...) SELECT * FROM site_access_copy PARTITION (p3, p4,...);
+```
+
+For detailed information such as the syntax and parameters descriptions, see [INSERT INTO](../sql-reference/sql-statements/data-manipulation/insert.md).
+
+### Load data by using STREAM LOAD
+
+Example:
+
+```undefined
+curl --location-trusted -u root: -H "label:123" -H "temporary_partitions: tp1, tp2, ..." -T testData \
+    http://host:port/api/example_db/site_access/_stream_load    
+```
+
+For detailed information such as the syntax and parameters descriptions, see [STREAM LOAD](..sql-reference/sql-statements/data-manipulation/STREAM%20LOAD.md).
+
+### Load data by using BROKER LOAD
+
+Example:
+
+```SQL
+LOAD LABEL example_db.label1
+(
+    DATA INFILE("hdfs://hdfs_host:hdfs_port/user/starrocks/data/input/file")
+    INTO TABLE my_table
+    TEMPORARY PARTITION (tp1, tp2, ...)
+    ...
+)
+WITH BROKER
+(
+    StorageCredentialParams
+);
+LOAD LABEL example_db.label1
+(
+    DATA INFILE("hdfs://hdfs_host:hdfs_port/user/starrocks/data/input/file")
+    INTO TABLE my_table
+    TEMPORARY PARTITION (tp1, tp2, ...)
+    ...
+)
+WITH BROKER "mybroker"
+(
+    "username" = "hdfs_username",
+    "password" = "hdfs_password"
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/edaec334-30d4-41bc-a577-45e89d6c9377)
From v2.5, no brokers are required in most scenarios. See Lines 138 to 141 which are correct.